onnxruntime/core/optimizer/double_qdq_pairs_remover.cc

```diff
@@ -82,19 +77,37 @@ bool DoubleQDQPairsRemover::IsNodeRemovable(
       grandchild->OpType() != "DequantizeLinear") {
     return false;
   }
+
   const auto get_constant_initializer = [&graph](const std::string& initializer_name) {
     return graph.GetConstantInitializer(initializer_name, true);
   };
+
+  // Each QDQ pair (i.e., parent -> self, child -> grandchild) has to meet the following additional requirements:
+  // - Scalar/constant zero-point and scale.
+  // - Zero-point input must exist.
```

Comment:
Why does the zero point input have to exist? Isn't it optional with a default of 0, so if missing can't we use the default value?

Reply:
I don't think the zero-point has to exist. As you say, a default value of 0 could be fine as long as it matches on both ops in a pair. In fact, it looks like this is the case everywhere we use `IsQDQPairSupported` (qdq_final_cleanup, qdq_selectors::DropQDQNodeGroupSelector::Check).

I added a TODO to address this in a future PR, since this would require changes not directly related to this PR.

Reply:
I reverted all the changes to this file as they are not needed for this PR. I'll revisit in the next PR to add 16-bit support to QDQ ops.